src/components/bookListItem.component.js

```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import store from '../store';
+import actions from '../actions'
+
+const changeSelectedBook = (book) => {
+    store.dispatch(actions.selectBookAction(book))
```

Comment:
We do not call store directly, instead we want the call to look just like `selectBookAction(book)`,
As repeating extra code lines for the dispatching are not easily readable, specially when we need a lot of actions.

So, to achieve this,
We need to connect that component, like other components above, except that connect statement would look like : 
```
import { bindActionCreators } from 'redux'
import actions from '../actions'
const mapDispatchToProps = function (dispatch) {
  // now we will have a prop called 'selectBookAction' that when called,
  // actions.selectBookAction will be called,
  // and it's return will be passed through store to all reducers to update redux state.
  return bindActionCreators({ selectBookAction: actions.selectBookAction }, dispatch)
}
// mapStateToProps is replaced with null, As we do not redux state here
// second argument is used to provide access to redux actions inside component
export default connect(null, mapDispatchToProps)(BookListItem)
```

Reply:
@lebaz20 is this also right ? ``` import React from 'react';
import store from '../store';
import actions from '../actions'

const BookListItem = ({book}) => {

    return (
        <div className="BookListItem">
            <div className="card" onClick={() => {
               store.dispatch(actions.selectBookAction(book))

            }}>
                <div className="card-body">
                    <h5 className="card-title">{book.name}</h5>
                </div>
            </div>
        </div>
    )
}


export default BookListItem
 ```